Framework/PythonInterface/plugins/algorithms/GSASIIRefineFitPeaks.py

```diff
@@ -15,974 +16,227 @@ class GSASIIRefineFitPeaks(PythonAlgorithm):
-
-    def _get_focused_wks(self, wks_prop_name, index_prop_name):
-        in_wks = self.getProperty(wks_prop_name).value
-        in_idx = self.getProperty(index_prop_name).value
-
-        if in_wks.getNumberHistograms() > 1 or 0 != in_idx:
-            focused_wks = msapi.ExtractSpectra(InputWorkspace = in_wks, StartWorkspaceIndex = in_idx,
-                                               EndworkspaceIndex = in_idx)
-        else:
-            focused_wks = in_wks
-
-        return focused_wks
-
-    def _load_prepare_data_for_fit(self, gs2, gs2_focused_wks, inst_file):
-        """
-        Loads the data into an "rd" object as used in the GSAS-II python modules, and
-        prepares an initia peaks list and a background function definition.
-        Importantly, this reads the rd.powderdata list with the histogram data,
-        and the rd.pwdparms tuple with instrument parameter lists
-
-        @param gs2 :: the main GSAS-II object
-        @param gs2_focused_wks :: a focused (single spectrum) workspace
-        @param inst_file :: GSAS instrument parameters file (.par / .prm / .iparm, etc.)
-
-        @returns a tuple with: 1) 3) GSAS-II "rd" object. 2) limits for fitting,
-        3) list of peaks, 4) background definition, These are ready to be passed to the
-        peak fitting functions
-        """
-        gs2_rd = self._build_gsas2_reader_with_data(gs2, gs2_focused_wks)
-        self.log().information("Loaded histogram data in GSAS-II data object: {0}".
-                               format(gs2_rd.powderdata))
+        refinement_method = self.getPropertyValue(self.PROP_REFINEMENT_METHOD)
+        if refinement_method == self.REFINEMENT_METHODS[2]:  # Peak fitting
+            raise NotImplementedError("GSAS-II Peak fitting not yet implemented in Mantid")
 
-        gs2_rd, inst_parms = self._add_instrument_info(gs2, gs2_rd, inst_file)
-        self.log().information("Parameters from instrument file: {0}".format(gs2_rd.pwdparms))
+        with self._suppress_stdout():
+            gsas_proj = self._initialise_GSAS()
 
-        background_def = self._build_add_background_def(gs2_rd)
-        self.log().information("Using background function: {0}".format(background_def))
+            rwp, gof, lattice_params = \
+                self._run_rietveld_pawley_refinement(gsas_proj=gsas_proj,
+                                                     do_pawley=refinement_method == self.REFINEMENT_METHODS[0])
+            self._set_output_properties(rwp=rwp, gof=gof, lattice_params=lattice_params)
 
-        limits = self._build_add_limits(gs2_rd)
-        self.log().information("Fitting loaded histogram data, with limits: {0}".format(limits))
-
-        # self.PROP_PHASE_INFO_FILE - phase information into rd, this is loaded elsewhere when
-        # doing Pawley/Rietveld refinement
+    def _build_output_lattice_table(self, lattice_params):
+        alg = self.createChildAlgorithm('CreateEmptyTableWorkspace')
```

Comment:
Does this need the `StoreInADS=False` thing now?

Reply:
I don't think so - child algorithms don't store in the ADS anyway, right? I thought that `StoreInADS` was for calling algorithms through `simpleapi`